Feature/pt 2/theme and navigation

flutterVlad opened the pull request:
https://github.com/flutterVlad/food_app/assets/137285738/ba151c9a-552a-47f4-b6ce-1c6889ae070f

flutterVlad committed:
[PT-2] feature/PT-2/theme-and-navigation

- Added 4 new packages:
  - setting
  - order_history
  - cart
  - product_detailed_view
- Added Light and Dark theme and switching them
- Added navigation between routes
- Added detail view about dish

flutterVlad committed:
[PT-2] feature/PT-2/theme-and-navigation

- Added 4 new packages:
  - setting
  - order_history
  - cart
  - product_detailed_view
- Added Light and Dark theme and switching them
- Added navigation between routes
- Added detail view about dish

+

- Little debugging

AliaksandrHuzich commented on core_ui/lib/src/theme/app_theme.dart:
```diff
@@ -2,23 +2,80 @@ import 'package:flutter/material.dart';
+            fontSize: 30,
+            color: Colors.black,
+            fontFamily: 'Dosis',
+          )),
```
,

AliaksandrHuzich commented on core_ui/lib/src/widgets/app_bar.dart:
```diff
@@ -0,0 +1,30 @@
+      : preferredSize = const Size.fromHeight(kToolbarHeight),
+        super(key: key);
+
+  static const Map<String, String> appBarTitle = {
```
костыль

AliaksandrHuzich commented on core_ui/lib/src/widgets/gradient_widget.dart:
```diff
@@ -0,0 +1,23 @@
+    required this.child,
+  }) : super(key: key);
+
+  final Widget child;
```
move to top

AliaksandrHuzich commented on core_ui/lib/src/widgets/gradient_widget.dart:
```diff
@@ -0,0 +1,23 @@
+import 'package:flutter/material.dart';
+
+class GradientWidget extends StatelessWidget {
```
Widget remove (rename)

AliaksandrHuzich commented on core_ui/lib/src/widgets/scaffold_skeleton.dart:
```diff
@@ -0,0 +1,55 @@
+import 'package:flutter/material.dart';
```
rename file

AliaksandrHuzich commented on features/home/lib/src/bloc/home/bloc.dart:
```diff
@@ -18,21 +18,15 @@ class ProductBloc extends Bloc<ProductEvent, ProductState> {
-
   void _loadProductList() async {
     emit(LoadingState());
     products = await _getAllProductsUseCase.execute(const NoParams());
```
remove these variables from bloc, use only state

AliaksandrHuzich commented on features/home/lib/src/ui/home_screen.dart:
```diff
@@ -22,89 +25,71 @@ class HomeScreenState extends State<HomeScreen> {
-                    return Column(
+      child: BlocBuilder<ProductBloc, ProductState>(
+        builder: (BuildContext context, ProductState state) {
+          final List<ProductModel> products = state.getProducts;
```
remove these

AliaksandrHuzich commented on features/home/lib/src/ui/home_screen.dart:
```diff
@@ -22,89 +25,71 @@ class HomeScreenState extends State<HomeScreen> {
+            return const AppCenterLoader();
+          }
+          if (state is LoadedState) {
+            return ListView.builder(
```
rewrite with grid

AliaksandrHuzich commented on features/home/lib/src/widgets/advertisement_block.dart:
```diff
@@ -1,30 +1,44 @@
+            horizontal: 20,
+          ),
+          decoration: BoxDecoration(
+            gradient: state == AppTheme.getLightThemeData()
```
change it

AliaksandrHuzich commented on features/home/lib/src/widgets/home_card_element.dart:
```diff
@@ -1,71 +1,92 @@
+                  builder: (context, state) {
+                    return Container(
+                      width: int.parse(
+                              '${MediaQuery.of(context).size.width.round()}') *
```
change it and below

AliaksandrHuzich commented on features/product_detailed_view/lib/src/ui/product_detailed_view.dart:
```diff
@@ -0,0 +1,92 @@
+        final Color containerColor = Theme.of(context).primaryColor;
+        return Container(
+          margin: const EdgeInsets.symmetric(horizontal: 15),
+          child: ListView.builder(
```
grid

AliaksandrHuzich commented on features/product_detailed_view/lib/src/ui/product_detailed_view.dart:
```diff
@@ -0,0 +1,92 @@
+                      decoration: BoxDecoration(
+                          borderRadius: BorderRadius.circular(30),
+                          color: containerColor,
+                          boxShadow: [
```
type

AliaksandrHuzich commented on features/product_detailed_view/lib/src/ui/product_detailed_view.dart:
```diff
@@ -0,0 +1,92 @@
+                              spreadRadius: 7,
+                              blurRadius: 7,
+                            ),
+                          ]),
```
,

AliaksandrHuzich commented on features/settings/lib/src/bloc/theme/bloc.dart:
```diff
@@ -0,0 +1,30 @@
+  }
+
+  void _init(InitialThemeSetEvent event, Emitter emit) async {
+    final bool hasDarkTheme = await isDark();
```
usecase here

AliaksandrHuzich commented on features/settings/lib/src/bloc/theme/bloc.dart:
```diff
@@ -0,0 +1,30 @@
+  }
+
+  void _switchTheme(ThemeSwitchEvent event, Emitter emit) {
+    final isDark = state == AppTheme.getDarkThemeData();
```
type

AliaksandrHuzich commented on features/settings/lib/src/bloc/theme/bloc.dart:
```diff
@@ -0,0 +1,30 @@
+    on<ThemeSwitchEvent>(_switchTheme);
+  }
+
+  void _init(InitialThemeSetEvent event, Emitter emit) async {
```
Future<void>

AliaksandrHuzich commented on features/settings/lib/src/bloc/theme/theme_helper.dart:
```diff
@@ -0,0 +1,11 @@
+part of 'bloc.dart';
+
```
remove this and add to data

AliaksandrHuzich commented on features/settings/lib/src/ui/setting_screen.dart:
```diff
@@ -0,0 +1,43 @@
+                builder: (BuildContext context, ThemeData state) {
+                  return Switch(
+                    value: state == AppTheme.getLightThemeData(),
+                    onChanged: (bool val) {
```
_ instead of val

AliaksandrHuzich reviewed (approved):
Data & domain layer, grid, типы местами и нейминг

flutterVlad committed:
[PT-2] feature/PT-2/theme-and-navigation

- Added 4 new packages:
  - setting
  - order_history
  - cart
  - product_detailed_view
- Added Light and Dark theme and switching them
- Added navigation between routes
- Added detail view about dish

+

- Little debugging

+

- Added SharedPreferences into the Data and the Domain layers and into the DI
- Changed ListView to GridView on start screen
- Fixed little bugs

flutterVlad committed:
[PT-2] feature/PT-2/theme-and-navigation

- Added 4 new packages:
  - setting
  - order_history
  - cart
  - product_detailed_view
- Added Light and Dark theme and switching them
- Added navigation between routes
- Added detail view about dish

+

- Little debugging

+

- Added SharedPreferences into the Data and the Domain layers and into the DI
- Changed ListView to GridView on start screen
- Fixed little bugs
